File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:

  void right(TreeNode* temp,vector<int>&arr,int level){
    if(temp == NULL)return;
    if(level == arr.size()) arr.push_back(temp->val);
    right(temp->right,arr,level+1);
    right(temp->left,arr,level+1);

  }
    vector<int> rightSideView(TreeNode* root) {
        vector<int>arr;
        right(root,arr,0);
        return arr;
    }
};

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    void help(TreeNode root , int lvl, List<Integer> ls){
        if(root == null) return;

        if(ls.size() == lvl){
            ls.add(root.val);
        }
        help(root.right , lvl+1, ls);
        help(root.left, lvl+1, ls);

    }
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> ls = new ArrayList<>();
        help(root, 0 , ls);
        return ls;

    }
}